File: java/src/Arrays/ValidAnagram.java
package Arrays;

public class ValidAnagram {
    public static boolean solution(String s, String t) {
        if (s.length() != t.length()) return false;
        int[] alphabet = new int[26];
        for (char c : s.toCharArray())
            alphabet[c - 'a']++;
        for (char c : t.toCharArray())
            alphabet[c - 'a']--;
        for (int i = 0; i < alphabet.length; i++) {
            if (alphabet[i] != 0) return false;
        }
        return true;
    }

    public static boolean solutionTwo(String s, String t) {

        if (s.length() != t.length()) return false;

        int[] charHash = new int[26];

        for (int i = 0; i < s.length(); i++) {
            charHash[s.charAt(i) - 'a'] += 1;
            charHash[t.charAt(i) - 'a'] -= 1;
        }
        for (int i : charHash) if (i != 0) return false;
        return true;
    }
}

File: csharp/Arrays/ValidAnagram.cs
namespace csharp.Arrays;

public class ValidAnagram
{
    public static bool Solution(string s, string t)
    {
        if (s.Length != t.Length) return false;
        int[] charArray = new int[26];
        for (int i = 0; i < s.Length; i++)
        {
            charArray[s[i] - 'a']++;
            charArray[t[i] - 'a']--;
        }
        //
        // Console.WriteLine("[{0}]", String.Join(", ", charArray));
        foreach (var k in charArray)
        {
            if (k != 0) return false;
        }
        return true;
    }
}